src/libraries/Microsoft.Extensions.Caching.Memory/tests/CompactTests.cs

```diff
@@ -82,5 +84,54 @@ public void CompactPrioritizesLRU()
             Assert.Equal(1, cache.Count);
             Assert.Equal("value4", cache.Get("key4"));
         }
+
+        /// <summary>
+        /// Tests a race condition in Compact where CacheEntry.LastAccessed is getting updated
+        /// by a different thread than what is doing the Compact, leading to sorting failing.
+        /// 
+        /// See https://github.com/dotnet/runtime/issues/61032.
+        /// </summary>
+        [Fact]
+        public void CompactLastAccessedRaceCondition()
+        {
+            const int numEntries = 100;
+            MemoryCache cache = new MemoryCache(new MemoryCacheOptions());
+            Random random = new Random();
+
+            void FillCache()
+            {
+                for (int i = 0; i < numEntries; i++)
+                {
+                    cache.Set($"key{i}", $"value{i}");
+                }
+            }
+
+            // start a few tasks to access entries in the background
+            Task[] backgroundAccessTasks = new Task[Environment.ProcessorCount];
```

Comment:
Is there a way to say it just for this test? Looking at https://xunit.net/docs/running-tests-in-parallel, it looks like the only options are to use `Collections` or to turn it off for the whole assembly.

Reply:
> the only options are to use Collections or to turn it off for the whole assembly.

You are right, this can be done only on class or assembly level. 

I've been refactoring the code around it this week (#62132). To disable it per type now you need to reference it in the test project:

```xml
<Compile Include="$(CommonTestPath)TestUtilities\System\DisableParallelization.cs" />
```

And use an attribute: `[Collection(nameof(DisableParallelization))]`

Reply:
Using that will only disable running this test in parallel with other tests in the same collection, right? There's no way of saying "run this one test by itself without any other test running at the same time" AFAIK.

Reply:
> There's no way of saying "run this one test by itself without any other test running at the same time"

I think that using `[CollectionDefinition(DisableParallelization = true)]` is exactly that.

According to [xUnit docs](https://xunit.net/docs/running-tests-in-parallel.html):

> Turn off parallelism for specific Test Collection
  [CollectionDefinition(DisableParallelization = true)]
Default: false
Parallel-capable test collections will be run first (in parallel), followed by parallel-disabled test collections (run sequentially).

Reply:
@eerhardt xUnit defnitely does not make it easy for us ;)

Reply:
Ah, I got it now. Thanks for the explanation. If I pull this test into its own class with that attribute on it, then all the other tests will run in parallel, and after that, this test will run by itself.